app/javascript/components/popup/UpdateUsername.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <el-dialog 
+    v-model="getChangeUsername" 
+    :title="$t('profile.edit.updateUsername')" 
+    width="30%" 
+    class="dialogWidth"
+    style="border-radius: 0.5rem" 
+    left
+  >
+    <el-form :model="formData" :rules="formRules" ref="formRef">
+      <div>
+        <div class="flex">
+          <SvgIcon name="exclamation_point" width="20px" height="20px" />
+          <p class="text-gray-500 text-[16px] ml-[8px]">
+            {{ $t('profile.edit.confirmUpdateMessage')}}
+          </p>
+        </div>
+        <p class="text-[#303133] text-[14px] mb-[8px] mt-[8px]">
+          {{ $t('profile.edit.username') }}
+          <span class="text-red-400">*</span>
+        </p>
+        <el-form-item prop="username">
+          <el-input v-model="formData.username" :rows="6" placeholder="请输入用户名" />        
+        </el-form-item>
+        <p class="text-gray-500 text-[12px] italic mt-[24px]">
+          {{ $t('rule.nameRule') }}
+        </p>
+      </div>
+    </el-form>
+    <template #footer>
+      <span class="dialog-footer">
+        <el-button @click="canChangeUsername = 'false'">{{ $t('all.cancel') }}</el-button>
+        <el-button type="primary" @click="submiteUsername">
+          {{ $t('all.update') }}
+        </el-button>
+      </span>
+    </template>
+  </el-dialog>
+</template>
+
+<script setup>
+  import { ref, inject, computed } from 'vue'
+  import { useI18n } from 'vue-i18n'
+  import { useCookies } from 'vue3-cookies'
+  import jwtFetch from '../../packs/jwtFetch.js'
+  import { ElMessage } from 'element-plus'
+
+  const { cookies } = useCookies()
+  const { t } = useI18n()
+  const csghubServer = inject('csghubServer')
+  const nameRule = inject('nameRule')
+  const formData = ref({ username: '' })
+  const formRef = ref(null)
+
+  const getChangeUsername = computed(() => {
+    const canChangeUsername = cookies.get('can_change_username')
+    return canChangeUsername == 'true'
```

Comment:
```suggestion
    return canChangeUsername === 'true'
```

Reply:
we should make sure the user is logged in, we can get this info from user Store

Reply:
done